### PR TITLE
create definition and mapper package and refactor toData

### PR DESCRIPTION
**Pull request by souissimai**

### CODE REVIEW

File: src/main/java/org/gridsuite/network/map/dto/mapper/battery/AbstractBatteryInfos.java [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+import java.util.List;
+import java.util.stream.Collectors;
+
+import static org.gridsuite.network.map.dto.utils.ElementUtils.nullIfNan;
+
+/**
+ * @author AJELLAL Ali <ali.ajellal@rte-france.com>
+ */
+
+@SuperBuilder
```
Comment (SlimaneAmar) [inline src/main/java/org/gridsuite/network/map/dto/mapper/battery/AbstractBatteryInfos.java:31]: Remove it

File: src/main/java/org/gridsuite/network/map/dto/mapper/battery/AbstractBatteryInfos.java [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+@Getter
+public abstract class AbstractBatteryInfos extends ElementInfos {
```
Comment (SlimaneAmar) [inline src/main/java/org/gridsuite/network/map/dto/mapper/battery/AbstractBatteryInfos.java:33]: Rename by `BatteryInfosMapper` and remove the `extends`

File: src/main/java/org/gridsuite/network/map/dto/mapper/battery/AbstractBatteryInfos.java [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+        switch (dataType) {
+            case TAB:
+                return toBatteryTabInfos(identifiable);
+            case FORM:
+                return toBatteryFormInfos(identifiable);
+            default:
+                throw new UnsupportedOperationException("TODO");
+        }
+    }
+
```
Comment (SlimaneAmar) [inline src/main/java/org/gridsuite/network/map/dto/mapper/battery/AbstractBatteryInfos.java:44]: public static ElementInfos toData(Identifiable<?> identifiable, ElementInfos.InfoType dataType) {
        return switch (dataType) {
            case TAB -> toBatteryTabInfos(identifiable);
            case FORM -> toBatteryFormInfos(identifiable);
            default -> throw new UnsupportedOperationException("TODO");
        };
    }

File: src/main/java/org/gridsuite/network/map/dto/mapper/battery/AbstractBatteryInfos.java [hunk 4]
```diff
@@ -0,0 +1,147 @@
+/**
+ * Copyright (c) 2023, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package org.gridsuite.network.map.dto.mapper.battery;
```
Comment (SlimaneAmar) [inline src/main/java/org/gridsuite/network/map/dto/mapper/battery/AbstractBatteryInfos.java:7]: Remove package battery

**Review by SlimaneAmar — CHANGES_REQUESTED** [review]

File: src/main/java/org/gridsuite/network/map/dto/mapper/battery/AbstractBatteryInfos.java [hunk 5]
```diff
@@ -0,0 +1,147 @@
...
+
+@SuperBuilder
+@Getter
+public abstract class AbstractBatteryInfos extends ElementInfos {
+    public static ElementInfos toData(Identifiable<?> identifiable, InfoType dataType) {
+        switch (dataType) {
+            case TAB:
+                return toBatteryTabInfos(identifiable);
+            case FORM:
+                return toBatteryFormInfos(identifiable);
```
Comment (SlimaneAmar) [inline src/main/java/org/gridsuite/network/map/dto/mapper/battery/AbstractBatteryInfos.java:39]: Rename by toFormInfos

**Comment by SlimaneAmar** [comment]
Make all methods `private` in the mappers

File: src/main/java/org/gridsuite/network/map/dto/mapper/battery/AbstractBatteryInfos.java (hunk 3 above)
Comment (souissimai) [reply src/main/java/org/gridsuite/network/map/dto/mapper/battery/AbstractBatteryInfos.java]: this format is not supported by checkSyle

File: src/main/java/org/gridsuite/network/map/dto/mapper/battery/AbstractBatteryInfos.java (hunk 1 above)
Comment (souissimai) [reply src/main/java/org/gridsuite/network/map/dto/mapper/battery/AbstractBatteryInfos.java]: Done

File: src/main/java/org/gridsuite/network/map/dto/mapper/battery/AbstractBatteryInfos.java (hunk 4 above)
Comment (souissimai) [reply src/main/java/org/gridsuite/network/map/dto/mapper/battery/AbstractBatteryInfos.java]: Done

File: src/main/java/org/gridsuite/network/map/dto/mapper/battery/AbstractBatteryInfos.java (hunk 2 above)
Comment (souissimai) [reply src/main/java/org/gridsuite/network/map/dto/mapper/battery/AbstractBatteryInfos.java]: Done

File: src/main/java/org/gridsuite/network/map/dto/mapper/battery/AbstractBatteryInfos.java [hunk 6]
```diff
@@ -0,0 +1,147 @@
...
+ * @author AJELLAL Ali <ali.ajellal@rte-france.com>
+ */
+
+@SuperBuilder
+@Getter
+public abstract class AbstractBatteryInfos extends ElementInfos {
+    public static ElementInfos toData(Identifiable<?> identifiable, InfoType dataType) {
+        switch (dataType) {
+            case TAB:
+                return toBatteryTabInfos(identifiable);
```
[the remark this answers is not in the document]
Comment (souissimai) [reply src/main/java/org/gridsuite/network/map/dto/mapper/battery/AbstractBatteryInfos.java]: Done

File: src/main/java/org/gridsuite/network/map/dto/mapper/battery/AbstractBatteryInfos.java (hunk 5 above)
Comment (souissimai) [reply src/main/java/org/gridsuite/network/map/dto/mapper/battery/AbstractBatteryInfos.java]: Done

**Comment by souissimai** [comment]
> Make all methods `private` in the mappers

some methods are called in other classes so it should be protected or public if it's not on the same package

File: src/main/java/org/gridsuite/network/map/dto/definition/battery/BatteryFormInfos.java [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+
+import java.util.List;
+
+/**
+ * @author REHILI Ghazwa <ghazwa.rehili@rte-france.com>
+ */
+
+@SuperBuilder
+@Getter
+public class BatteryFormInfos extends AbstractBatteryInfos {
```
[the remark this answers is not in the document]
Comment (souissimai) [reply src/main/java/org/gridsuite/network/map/dto/definition/battery/BatteryFormInfos.java]: done

**Review by SlimaneAmar — COMMENTED** [review]
Replace all `getExtension(ConnectablePosition.class)` and do same for others extensions

**Review by SlimaneAmar — APPROVED** [review]

**Pull request merged**